components/org.wso2.carbon.identity.tag.manager.core/src/main/java/org/wso2/carbon/identity/tag/manager/core/dao/TagManagementDAOImpl.java

```diff
@@ -0,0 +1,361 @@
+import org.wso2.carbon.identity.tag.manager.model.TagListResult;
+import org.wso2.carbon.registry.core.utils.UUIDGenerator;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Tag management service DAO implementation class.
+ */
+public class TagManagementDAOImpl implements TagManagementDAO {
+
+    @Override
+    public Tag loadTag(String tagUuid, String tenantUuid) throws TagServiceException {
+
+        PreparedStatement prepStmt;
+        ResultSet resultSet;
+        Tag tag = null;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.GET_IDN_TAG_BY_TAG_UUID_SQL);
+            prepStmt.setString(1, tagUuid);
+            resultSet = prepStmt.executeQuery();
+            if (resultSet.next()) {
+                tag = new Tag();
+                tag.setTagUuid(resultSet.getString(1));
+                tag.setName(resultSet.getString(2));
+                tag.setDescription(resultSet.getString(3));
+                tag.setPubliclyVisible(isPublic(resultSet.getString(4)));
+                tag.setAssociationType(resultSet.getString(5));
+            }
+        } catch (SQLException e) {
+            throw new TagServiceException(ErrorMessage.ERROR_CODE_ERROR_LOADING_TAG.getCode(), e);
+        }
+        return tag;
+    }
+
+    @Override
+    public Tag loadTag(String name, String type, String tenantUuid) throws TagServiceException {
+
+        PreparedStatement prepStmt;
+        ResultSet resultSet;
+        Tag tag = null;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.GET_IDN_TAG_BY_NAME_TYPE_TENANT_UUID_SQL);
+            prepStmt.setString(1, name);
+            prepStmt.setString(2, type);
+            prepStmt.setString(3, tenantUuid);
+            resultSet = prepStmt.executeQuery();
+            if (resultSet.next()) {
+                tag = new Tag();
+                tag.setTagUuid(resultSet.getString(1));
+                tag.setDescription(resultSet.getString(2));
+                tag.setPubliclyVisible(isPublic(resultSet.getString(3)));
+            }
+        } catch (SQLException e) {
+            throw new TagServiceException(ErrorMessage.ERROR_CODE_ERROR_LOADING_TAG.getCode(), e);
+        }
+        return tag;
+    }
+
+    @Override
+    public TagListResult filterTags(int limit, int offset, String sortBy, String sortOrder, String filter,
+                                    String tenantUuid) throws TagServiceException {
+
+        if (limit == 0) {
+            return null;
+        }
+        PreparedStatement prepStmt;
+        ResultSet resultSet;
+        TagListResult tagListResult = new TagListResult();
+        List<Tag> tagList = new ArrayList<>();
```

Comment:
We can move this to the try-catch right?

Reply:
done